app/models/user.rb

```diff
@@ -575,6 +628,45 @@ def flipper_id
 
   private
 
+  def sync_relevant_profile_fields_to_user_settings_table(users_setting_record)
+    PROFILE_FIELDS_TO_MIGRATE_TO_USERS_SETTINGS_TABLE.each do |field|
+      users_setting_record.update(field => profile.data[field]) if profile.data[field].present?
+    end
+  end
+
+  def migrate_users_and_profile_fields_to_users_settings(users_setting_record)
+    USER_FIELDS_TO_MIGRATE_TO_USERS_SETTINGS_TABLE.each do |field|
+      if USER_SETTINGS_ENUM_FIELDS.include?(field)
+        field_enums = Users::Setting.defined_enums[field]
+        users_setting_record.update(field => field_enums[public_send(field).to_sym])
+      else
+        users_setting_record.update(field => public_send(field))
+      end
+    end
+
+    sync_relevant_profile_fields_to_user_settings_table(users_setting_record) if profile
+  end
+
+  def sync_users_settings_table
+    users_setting_record = Users::Setting.find_by(user_id: id)
+    users_setting_record ||= Users::Setting.create(user_id: id)
```

Comment:
this can be simplified with [create_or_find_by](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-create_or_find_by):

```suggestion
    users_setting_record = Users::Setting.create_or_find_by(user_id: id)
```